current.go

```diff
@@ -35,3 +38,55 @@ func (r *VaultRole) Exists(config ExistsConfig) (bool, error) {
 
 	return false, nil
 }
+
+type role struct {
+	AllowBareDomains bool   `json:"allow_bare_domains"`
+	AllowSubdomains  bool   `json:"allow_subdomains"`
+	AllowedDomains   string `json:"allowed_domains"`
+	Organizations    string `json:"organization"` // NOTE the singular form here.
+	TTL              string `json:"ttl"`
+}
+
+func (r *VaultRole) Search(config SearchConfig) (Role, error) {
+	// Check if a PKI for the given cluster ID exists.
+	secret, err := r.vaultClient.Logical().Read(key.ReadRolePath(config.ID, config.Organizations))
+	if IsNoVaultHandlerDefined(err) {
+		return Role{}, microerror.Maskf(notFoundError, "no vault handler defined")
+	} else if err != nil {
+		return Role{}, microerror.Mask(err)
+	}
+
+	// In case there is not a single role for this PKI backend, secret is nil.
+	if secret == nil {
+		return Role{}, microerror.Maskf(notFoundError, "no vault secret at path '%s'", key.RoleName(config.ID, config.Organizations))
+	}
+
+	b, err := json.Marshal(secret.Data)
+	if err != nil {
+		return Role{}, microerror.Mask(err)
+	}
+
+	var internalRole role
+	err = json.Unmarshal(b, &internalRole)
+	if err != nil {
+		return Role{}, microerror.Mask(err)
+	}
+
+	altNames := key.ToAltNames(internalRole.AllowedDomains)
+	organizations := key.ToOrganizations(internalRole.Organizations)
+	ttl, err := parseutil.ParseDurationSecond(internalRole.TTL)
+	if err != nil {
+		return Role{}, microerror.Mask(err)
+	}
+
+	newRole := Role{
+		AllowBareDomains: internalRole.AllowBareDomains,
+		AllowSubdomains:  internalRole.AllowSubdomains,
+		AltNames:         altNames,
+		ID:               config.ID,
+		Organizations:    organizations,
+		TTL:              ttl,
+	}
```

Comment:
We have to deal with our own type because of how Vault represents data and because of how we work with it. One detail is that Vault holds allowed domains but our interface is about alternative names which contain the common name of a guest cluster.